src/gleam/string.gleam

```diff
@@ -95,40 +95,87 @@ pub external fn uppercase(String) -> String = "string" "uppercase"
 pub external fn compare(String, String) -> order.Order =
   "gleam_stdlib" "compare_strings"
 
-/// ## Get Substrings
+// slice helper
+external fn erl_string_slice(string, start, length) -> String =
+  "string" "slice"
 
-// TODO
-/// Take a substring given a start and end Grapheme indexes. Negative indexes
-/// are taken starting from the *end* of the list.
+/// ## Get Substrings
+///
+/// Take a substring given a starting index and length.
+/// A negative start or length will return an empty string.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
-/// slice("gleam", from: 1, to: 3) == "lea"
-/// slice("gleam", from: 1, to: 10) == "leam"
-/// slice("snakes on a plane!", from: -6, to: -1) == "plane"
+/// slice("gleam", from: 1, length: 3) == "lea"
+/// slice("gleam", from: 1, length: 10) == "leam"
+/// slice("snakes on a plane!", from: -6, to: -1) == ""
 /// ```
 ///
-// pub fn slice(out_of string: String, from start: Int, end: Int) -> String {}
+pub fn slice(string: String, start start: Int, length length: Int) -> String {
+  case start < 0 || length <= 0 {
+    True ->
+      ""
+
+    False ->
+      erl_string_slice(string, start, length)
+  }
+
+}
 
-// TODO
-/// Drop *n* Graphemes from the left side of a
+// drop helpers
+external fn erl_string_slice_to_infinity(string: String, start: Int) -> String =
+    "string" "slice"
+
+/// Drop *n* Graphemes from the start of a string
+///
+/// If `up_to` is <= 0 the input string will be returned unchanged.
+/// If `up_to` is >= the input strings length, "" will be returned.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
 /// drop_left(from: "The Lone Gunmen", up_to: 2) == "e Lone Gunmen"
 /// ```
 ///
-// pub fn drop_left(from string: String, up_to num_graphemes: Int) -> String {}
+pub fn drop_left(from from: String, up_to up_to: Int) -> String {
```

Comment:
Lets use nouns for variable names.

Reply:
I don't have any good thoughts for this? You? Perhaps drop the name for the `from` var?